pyked/chemked.py

```diff
@@ -233,6 +241,238 @@ def get_dataframe(self, output_columns=None):
+        datagroup_link.set('dataPointID', '')
+
+        property_idx = {}
+        labels = {'temperature': 'T', 'pressure': 'P',
+                  'ignition delay': 'tau', 'pressure rise': 'dP/dt',
+                  }
+
+        for prop_name in datagroup_properties:
+            attribute = prop_name.replace(' ', '_')
+            if (prop_name not in common and
+                any([getattr(dp, attribute, None) for dp in self.datapoints])
+                ):
+                prop = etree.SubElement(datagroup, 'property')
+                prop.set('description', '')
+                prop.set('name', prop_name)
+                prop.set('units', str(getattr(self.datapoints[0], attribute).units))
+                idx = 'x{}'.format(len(property_idx) + 1)
+                property_idx[idx] = prop_name
+                prop.set('id', idx)
+                prop.set('label', labels[prop_name])
+
+        if 'composition' not in common:
+            for species in self.datapoints[0].composition:
+                prop = etree.SubElement(datagroup, 'property')
+                prop.set('description', '')
+
+                idx = 'x{}'.format(len(property_idx) + 1)
+                property_idx[idx] = species['species-name']
+                prop.set('id', idx)
+                prop.set('label', '[' + species['species-name'] + ']')
+                prop.set('name', 'composition')
+                prop.set('units', self.datapoints[0].composition_type)
+
+                species_link = etree.SubElement(prop, 'speciesLink')
+                species_link.set('preferredKey', species['species-name'])
+                if species.get('InChI'):
+                    species_link.set('InChI', species['InChI'])
+
+        for dp in self.datapoints:
+            datapoint = etree.SubElement(datagroup, 'dataPoint')
+            for idx in property_idx:
+                value = etree.SubElement(datapoint, idx)
+                # handle regular properties a bit differently than composition
+                if property_idx[idx] in datagroup_properties:
+                    value.text = str(getattr(dp, property_idx[idx].replace(' ', '_')).magnitude)
+                else:
+                    # composition
+                    for item in dp.composition:
+                        if item['species-name'] == property_idx[idx]:
+                            value.text = str(item['amount'].magnitude)
+
+        # if RCM and has volume history, need a second dataGroup
+        if (len(self.datapoints) > 1 and
+            any([getattr(dp, 'volume_history', None) for dp in self.datapoints])
+            ):
+            raise NotImplementedError('Error: ReSpecTh files do not support multiple datapoints '
+                                      'with a volume history.'
+                                      )
+            # TODO: what if they share the same history? Does this happen?
+        elif getattr(self.datapoints[0], 'volume_history', None):
```

Comment:
Again, consider `hasattr` for clarity. Also, should it be its own `if` rather than `elif`? This isn't really connected to the previous `if`